`General`: Add course title to enrollment confirmation

milljoniaer opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client

- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.


### Motivation and Context
This PR adds the course Title to the modal title of the Course Enrollment confirmation popup as requested by Stephan.

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
Prerequisites:
- 1 Student 
- 1 Course with active Course Enrollment and Confirmation Message 

1. Log in to Artemis
2. Navigate to Course Enrollment
3. Enroll for this course
4. Verify that there is the course title within the Modal title

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Screenshots
![image](https://github.com/ls1intum/Artemis/assets/78978542/e044d063-c815-4238-a8bd-addcfbc2fde8)

JohannesStoehr commented on src/main/webapp/app/overview/course-registration/course-registration-button/course-registration-button.component.ts:
```diff
@@ -17,11 +18,14 @@ export class CourseRegistrationButtonComponent implements OnInit {
     userIsAllowedToRegister = false;
     loading = false;
 
+    confirmationTitle: string;
```
Am I missing something? You only add this variable and initialize it, but I cannot see where you use its value

egekurt123 commented:
![Screenshot 2023-10-22 194508](https://github.com/ls1intum/Artemis/assets/69898283/72fca4b5-c70a-4a40-ac5d-f0b252891e25)
Currently, the pop-up does not showcase the course title

milljoniaer commented:
Hi, thank you for your reviews, I recognized that there is a better way to do this. 
So I changed the strategy, please have another look.

I also added a screenshot, who the modal should look like now.

jakubriegel reviewed (approved):
Code ok. Change works on TS4

sarpsahinalp reviewed (commented):
I wasn't able to see this feature on ts2 for the enrollment test course.
![image](https://github.com/ls1intum/Artemis/assets/92453703/e8c39443-aebe-4eb5-8f2a-29510e832dee)

On ts3 it seems to be working.
![image](https://github.com/ls1intum/Artemis/assets/92453703/c805bcc6-bb14-40f2-b3e8-8a302a81c07b)

JanaNF reviewed (approved):
Manual tested on legacy ts2. Worked as expected.

LenaKahle reviewed (approved):
Tested on ts4, works as intended. 
Initially I thought the green success message ("Enrollment successful") should contain the module name as well, but I guess it's intended only for the confirmation popup.
